test/beginend-dired-test.el

```diff
@@ -35,15 +35,42 @@
 
 (defun beginend-create-dired-buffer ()
   "Create and return a buffer ressembling a dired one."
-  (assess-with-filesystem '("dir1/" "dir2/")
+  (assess-with-filesystem ()
+    ;; Ensure dir1 is newer than the test directory.
+    (sleep-for 1)
```

Comment:
This slows down the test suite quite a bit. A faster way would be to use `(sleep-for 0 1)`. Another way to achieve the same would be to use `touch --date="..."` through `shell-command-to-string` or similar. Yet another way would be to use `dired-virtual` but I'm not sure we can do everything we need with that.

However, as expressed in another comment, I would like to get rid of all the tests below as our code is now completely independent of hide-details, omit or sorting. If you want, please try to get rid of the test suite and write something much simpler. The tests I have in mind are:

- make sure `beginend--dired-on-file-p` does not consider `.` or `..` as files
- make sure `beginend--dired-on-file-p` does not consider first lines with current directory name of size as files
- make sure going to end goes to `point-max` if the buffer does not contain any file

That would be enough for me I guess. You could even try `dired-virtual` for that.

If you don't feel for it, I will do it myself and only merge your changes on `beginend.el`.